### PR TITLE
[blocked] Add support for a timeout parameter for memory dumps

### PR DESCRIPTION
**Pull request by foursixnine**

### CODE REVIEW

File: testapi.pm [hunk 1]
```diff
@@ -1539,7 +1539,7 @@ sub eject_cd {
 
 =head2 save_memory_dump
 
-  save_memory_dump(filename => undef, migration_speed => "4096m");
+  save_memory_dump([$filename, $max_downtime]);
```
Comment (coolo) [inline testapi.pm]: this looks wrong. It's expecting a hash in my reading of the code

Comment (foursixnine) [reply testapi.pm]: You are correct :), it's wrong

File: testapi.pm [hunk 2]
```diff
@@ -1539,7 +1539,7 @@ sub eject_cd {
...
 
 Saves the SUT memory state using C<$filename> as base for the memory dump
 filename,  the default will be the current test's name.
```
Comment (okurz) [inline testapi.pm]: I recommend to mention the default `max_downtime` in the description. Yes, it's annoying that we have to repeat that but if we insist on using perl we are stuck with POD then IMHO we have to do that

**Comment by foursixnine** [comment]
So I have rebased locally, and found out that I'm getting:

`[Mon Dec  4 15:39:09 2017] [687:debug] capture loop failed panic: attempt to copy freed scalar 5579bc5a37b8 to 5579ba49c2d0 at /usr/lib/perl5/5.26.1/Carp.pm line 229.
`

**Comment by foursixnine** [comment]
see #892

**Comment by coolo** [comment]
hopefully no longer needed

**Pull request closed without merging**